lib/Upcoming.dart

```diff
@@ -0,0 +1,271 @@
+          leading: IconButton(
+            icon: Icon(Icons.arrow_back_ios, color: Colors.black),
+            onPressed: () => Navigator.pop(context, false),
+          ),
+        ),
+        body: Stack(
+          children: <Widget>[
+            ListView(
+              scrollDirection: Axis.vertical,
+              children: <Widget>[
+                Column(
+                  children: <Widget>[
+                    Row(
+                      children: <Widget>[
+                        Container(
+                          margin: EdgeInsets.only(left: 15, top: 15, bottom: 15),
+                          child: Text("Upcoming Ride",
+                              style: TextStyle(
+                                  color: Colors.black,
+                                  fontSize: 30,
+                                  fontFamily: 'SFPro',
+                                  fontWeight: FontWeight.bold)),
+                        )
+                      ],
+                    ),
+                    SizedBox(height: 10),
+                    Container(
+                      margin: EdgeInsets.only(left: 15),
+                      child: Row(
+                        children: <Widget>[
+                          Text('Ride Info', style: TextStyle(color: Colors.grey[800],
+                              fontSize: 20,
+                              fontFamily: 'SFPro',
+                              fontWeight: FontWeight.bold)),
+                          SizedBox(width: 10.0),
+                          Text('Confirmed',
+                              style: TextStyle(color: Colors.green[500], fontSize: 11))
+                        ],
+                      ),
+                    ),
+                    SizedBox(height: 20),
+                    Container(
+                      child: Row(
+                        children: <Widget>[
+                          SizedBox(width: 20),
+                          rideTimeLine(),
+                          SizedBox(width: 20),
+                          Container(
+                            height: 220,
+                            child: Column(
+                              children: <Widget>[
+                                informationRow('Upson Hall', '124 Hoy Rd, Ithaca, NY 14850', 'Pickup Aiden', '3:15 PM'),
+                                SizedBox(height: 20),
+                                informationRow('Gates Hall', '107 Hoy Rd, Ithaca, NY 14853', 'Pickup Aiden', '3:20 PM'),
+                                SizedBox(height: 20),
+                                informationRow('Statler Hall', '109 Tower Rd, Ithaca, NY 14850', 'Pickup Emma', '3:25 PM'),
+                                SizedBox(height: 20),
+                                informationRow('Rhodes Hall', '109 Tower Rd, Ithaca, NY 14850', 'Pickup Cory', '4:20 PM'),
+                              ],
+                            ),
+                          )
+                        ],
```

Comment:
Has overflow. Could you make this depend on the width of the screen? Also, remove the names of the riders because riders are not allowed to have the names of any other riders. Cornell takes privacy seriously.